dist/$rsi.schema.json

```diff
@@ -0,0 +1,224 @@
+{
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "title": "RSI Validation Schema",
+  "version": "2.0.0",
+  "$ref": "#/definitions/serviceDefinition",
+  "definitions": {
+    "serviceDefinition": {
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string",
+          "pattern": "^(viwi|rsi)\\.service\\.[a-zA-Z0-9]+$"
+        },
+        "resources": {
+          "$ref": "#/definitions/resourcesDefinition"
+        },
+        "types": {
+          "$ref": "#/definitions/typesDefinition"
+        },
+        "description": {
+          "type": "string"
+        },
+        "$schema": {
+          "type": "string"
+        }
+      },
+      "required": [
+        "description",
+        "name",
+        "$schema"
+      ],
+      "additionalProperties": false
+    },
+    "resourcesDefinition": {
+      "type": "object",
+      "properties": {
+        "description": {
+          "type": "string"
+        },
+        "objectModel": {
+          "$ref": "#/definitions/objectModelDefinition"
+        }
+      },
+      "required": [
+        "description"
+      ],
+      "additionalProperties": true
+    },
+    "typesDefinition": {
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string"
+        },
+        "description": {
+          "type": "string"
+        }
+      },
+      "required": [
+        "description"
+      ],
+      "additionalProperties": true
+    },
+    "objectModelDefinition": {
+      "type": "object",
+      "properties": {
+        "id": {
+          "type": "object",
+          "properties": {
+            "description": {
+              "type": "string",
+              "minLength": 2
+            },
+            "type": {
+              "type": "string",
+              "enum": [
+                "string",
+                "number",
+                "integer",
+                "boolean",
+                "array",
+                "object"
+              ]
+            },
+            "format": {
+              "type": "string",
+              "pattern": "uuid"
+            }
+          },
+          "required": [
+            "description",
+            "type",
+            "format"
+          ]
+        },
+        "name": {
+          "type": "string"
+        },
+        "uri": {
+          "type": "object",
+          "properties": {
+            "description": {
+              "type": "string",
+              "minLength": 2
+            },
+            "type": {
+              "type": "string",
+              "pattern": "string"
+            },
+            "format": {
+              "type": "string",
+              "pattern": "uri"
+            }
+          },
+          "required": [
+            "description",
+            "type",
+            "format"
+          ]
+        }
+      },
+      "patternProperties": {
+        "^[a-zA-Z0-9]+$": {
```

Comment:
the regex definition looks so different compared to the `objectModelReference`
https://github.com/wzr1337/rsi.schema/pull/1/files#diff-d04eb47088d1746cdef153bc7e71c9afR139

Reply:
Properties "id" may need to be reworked. "type" seems to be at a wrong level here.
Very good to include "required" here.